### PR TITLE
Update DataSourceLoadOptionsBase docs

### PR DESCRIPTION
**Pull request by AlekseyMartynov**

### CODE REVIEW

File: net/DevExtreme.AspNet.Data/DataSourceLoadOptionsBase.cs [hunk 1]
```diff
@@ -67,12 +67,21 @@ public class DataSourceLoadOptionsBase {
         /// </summary>
         public string[] Select { get; set; }
 
+        /// <summary>
+        /// An array of data fields that constrains the <see cref="Select" /> expression.
```
Comment (dirk-pieterse) [inline net/DevExtreme.AspNet.Data/DataSourceLoadOptionsBase.cs]: constrains? Explain.

File: net/DevExtreme.AspNet.Data/DataSourceLoadOptionsBase.cs [hunk 2]
```diff
@@ -67,12 +67,21 @@ public class DataSourceLoadOptionsBase {
...
+        /// The applied select expression is the intersection of <see cref="PreSelect" /> and <see cref="Select" />.
```
Comment (dirk-pieterse) [inline net/DevExtreme.AspNet.Data/DataSourceLoadOptionsBase.cs]: applied select expression? Perhaps you should indicate the 'select' is the name of the expression?
"is the intersection of" Explain.

File: net/DevExtreme.AspNet.Data/DataSourceLoadOptionsBase.cs [hunk 3]
```diff
@@ -67,12 +67,21 @@ public class DataSourceLoadOptionsBase {
...
+        /// </summary>
         public string[] PreSelect { get; set; }
 
+        /// <summary>
+        /// A flag that indicates whether the select expression must be executed by the LINQ provider.
```
Comment (dirk-pieterse) [inline net/DevExtreme.AspNet.Data/DataSourceLoadOptionsBase.cs]: A flag that indicates whether the select expression must be executed by the LINQ provider. -> A flag that indicates whether the LINQ provider should execute the **select** expression.

File: net/DevExtreme.AspNet.Data/DataSourceLoadOptionsBase.cs [hunk 4]
```diff
@@ -67,12 +67,21 @@ public class DataSourceLoadOptionsBase {
...
+        /// If set to false, the select operation is performed in memory.
+        /// </summary>
         public bool? RemoteSelect { get; set; }
 
         /// <summary>
-        /// A flag that indicates whether grouping parameters must be translated into LINQ.
+        /// A flag that indicates whether grouping must be executed by the LINQ provider.
```
Comment (dirk-pieterse) [inline net/DevExtreme.AspNet.Data/DataSourceLoadOptionsBase.cs]: A flag that indicates whether grouping must be executed by the LINQ provider. -> A flag that indicates whether the LINQ provider should execute grouping.

File: net/DevExtreme.AspNet.Data/DataSourceLoadOptionsBase.cs (hunk 1 above)
Comment (AlekseyMartynov) [reply net/DevExtreme.AspNet.Data/DataSourceLoadOptionsBase.cs]: Say there are 100 fields in the data source. Without `PreSelect`, it's possible to use any of those 100 in `Select`.
With `PreSelect`, the developer fixes/constrains fields that are allowed for usage in `Select`. Not allowed fields specified in `Select` are ignored without errors.

**Pull request merged**